### PR TITLE
[Core][Chaos Test] Chaos testing fine-tuning mocking workload

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR adds a fine-tuning mocking workload in chaos testing CI tests. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Seems like it fails with the same error as endpoint. cc @rynewang I will merge this after you merge that PR

**Comment by rynewang** [comment]
Good test! Maybe we can merge this after the matrix testing fix as a new workload?

**Comment by rkooo567** [comment]
@rynewang I feel like it is blocked too long. If the other PR is not merged by tomorrow, I will probably just merge it first (so that we can run chaos tests asap). After we merge your PR, I can work further to use the same pattern

File: python/ray/tests/chaos/fine_tuning_like.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            for ref in refs:
+                finished = await ref
+                if not finished:
+                    done = False
+            if done:
+                break
+
+
+@ray.remote(num_cpus=0)
+class FakeFineTuninig:
```
Comment (rynewang) [inline python/ray/tests/chaos/fine_tuning_like.py:60]: I wonder if it's possible we use Ray Tune with a fake tune load? something like (example from https://docs.ray.io/en/latest/tune/index.html):

```
from ray import train, tune


def objective(config):  # ①
    score = config["a"] ** 2 + config["b"]
    return {"score": score}


search_space = {  # ②
    "a": tune.grid_search([0.001, 0.01, 0.1, 1.0]),
    "b": tune.choice([1, 2, 3]),
}

tuner = tune.Tuner(objective, param_space=search_space)  # ③

results = tuner.fit()
print(results.get_best_result(metric="score", mode="min").config)
```

File: .buildkite/pipeline.build.yml [hunk 2]
```diff
@@ -318,6 +318,56 @@
     - echo "--- Running the script 100 times with fault of networking delay"
     - for i in {1..100}; do ray job submit --address http://localhost:8265 --runtime-env python/ray/tests/chaos/runtime_env.yaml --working-dir python/ray/tests/chaos -- python potato_passer.py --num-actors=3 --pass-times=3 --sleep-secs=0.01; done
 
+- label: ":kubernetes: :mending_heart: :runner: chaos network delay fine-tuning-mocking-workload"
```
Comment (rynewang) [inline .buildkite/pipeline.build.yml:321]: We can reuse the CI matrix PR here

File: t.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import asyncio
+def sigterm_handler():
+    logger.warning("Exiting with SIGTERM immediately...")
+    # Exit code 0 will be considered as an expected shutdown
+    os._exit(signal.SIGTERM)
+async def run():
+    await asyncio.sleep(3600)
+
+loop = asyncio.get_event_loop()
+loop.run_until_complete(run())
```
Comment (rynewang) [inline t.py:12]: debugging leftover? Also do we need that serve generated file?

File: python/ray/tests/chaos/fine_tuning_like.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/tests/chaos/fine_tuning_like.py:60]: actually it is not a tune test. It is "fine tuninig" (which means you run training workload on top of existing weights). Re; using library directly, I prefer to just minimze the lib dependencies

File: t.py (hunk 3 above)
Comment (rkooo567) [reply t.py]: will remove them

**Comment by rkooo567** [comment]
cc @rynewang to merge

**Review by rynewang — APPROVED** [review]

**Comment by rkooo567** [comment]
Hmm this seems pretty difficult to pass in the small environment like this (with high memory constraint). I think we should just add this one to release tests

**Pull request closed without merging**